voip_utils/sip.py

```diff
@@ -364,6 +560,38 @@ def _parse_sip(
 
         return method, ruri, headers, body
 
+    def _parse_sip_reply(
+        self, message: str
+    ) -> Tuple[Optional[str], Optional[str], Optional[str], Dict[str, str], str]:
+        """Parse SIP message and return method, headers, and body."""
+        lines = message.splitlines()
+
+        protocol: Optional[str] = None
+        code: Optional[str] = None
+        reason: Optional[str] = None
+        headers: dict[str, str] = {}
+        offset: int = 0
+
+        # See: https://datatracker.ietf.org/doc/html/rfc3261
+        for i, line in enumerate(lines):
+            if line:
+                offset += len(line) + len(_CRLF)
+
+            if i == 0:
+                line_parts = line.split()
+                protocol = line_parts[0]
+                code = line_parts[1]
+                reason = line_parts[2]
+            elif not line:
+                break
+            else:
+                key, value = line.split(":", maxsplit=1)
+                headers[key.lower()] = value.strip()
+
+        body = message[offset:]
+
+        return protocol, code, reason, headers, body
```

Comment:
Let's use a dataclass for this. And let's make the parsing a static method on it.